lib/Graph/TensorLayout.cpp

```diff
@@ -602,12 +602,16 @@ bool TensorLayoutCommon::isSatisfiedBy(
   return true;
 }
 
-static std::string returnBaseReqOrNHWC(std::string baseReq) {
+static std::string returnBaseReqOrNHWC(std::string baseReq, const Node *node) {
   auto baseReqHelper = TensorLayoutDescription(baseReq);
   if (!baseReqHelper.isSameLayout(
           CanonicalTensorLayout::getInstance().getLayoutsForDims()[4])) {
     return baseReq;
   }
+  if (CanonicalTensorLayout::getInstance().acceptsAnyLayout(node)) {
+    // These nodes accept any 4-D layout.
```

Comment:
Is it important that the node has specifically a 4-D layout?

Reply:
yes. the function is called `returnBaseReqOrNHWC` and in canonical mode only 4-D got special requirements